### PR TITLE
Add ember-animated types

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: types/ember-animated/animated-if.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    predicate: any;
+    duration?: number;
+  };
+  Yields: {
+    default: [];
+    inverse?: [];
+  };
+}
+
+export declare class AnimatedIf extends Component<AnimatedIfSignature> {}
```
[the remark this answers is not in the document]
Comment (wagenet) [reply types/ember-animated/animated-if.d.ts:14]: I'm using this in my own app and it works fine.

**Comment by wagenet** [comment]
I realized that I should add support for positional args.

File: types/ember-animated/animated-each.d.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+export default class AnimatedEach<T> extends Component<AnimatedEachSignature<T>> {}
+
+interface AnimatedEachCurlySignature<T> {
+  Args: AnimatedEachArgs;
+  PositionalArgs: [
+    /* The list of data you are trying to render. */
+    items: T[]
+  ];
+  Yields: {
```
Comment (bakerac4) [inline types/ember-animated/animated-each.d.ts:56]: @wagenet it seems like we are duplicating the yields here. Thoughts on either making a `AnimatedEachYields` to reduce duplication or do what I did with the `ModifyYields` helper to make a `CurlySignature` util that could be used in the rest of the files as well

```

export type CurlySignature<T, Y> = Omit<T, 'Args'> & Y;

interface AnimatedEachCurlyArgs<T> {
    Args: AnimatedEachArgs;
    PositionalArgs: [
        /* The list of data you are trying to render. */
        items: T[]
    ];
}

export class AnimatedEachCurly<T> extends Component<CurlySignature<AnimatedEachSignature<T>, AnimatedEachCurlyArgs<T>>> {}
```

File: types/ember-animated/animated-each.d.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  key?: string;
+
+  /** Specifies data-dependent Rules that choose which Transition to run when the list changes. This takes precedence over use. */
+  rules?: AnimatedEachComponent['rules'];
+
+  /** Specifies the Transition to run when the list changes. */
+  use: Transition;
+
+  /** An optional comma-separated list of properties to observe on each of the objects in the items list. If any of those properties change, we will trigger an animated transition. Without this, we only animate when the list contents change, not when any deeper properties change. */
+  watch: String;
```
Comment (bakerac4) [inline types/ember-animated/animated-each.d.ts:34]: Did you mean to use capital `String` here?

Comment (wagenet) [reply types/ember-animated/animated-each.d.ts:34]: Nope, bad cut and paste.

File: types/ember-animated/animated-each.d.ts (hunk 2 above)
Comment (wagenet) [reply types/ember-animated/animated-each.d.ts:56]: Good idea.

**Review by bakerac4 — APPROVED** [review]

**Pull request merged**